cores/esp8266/WString.h

```diff
@@ -211,10 +211,9 @@ class String {
             return *this;
         }
 
-        // checks whether the internal buffer pointer is set.
-        // (should not be the case for us, since we always reset the pointer to the SSO buffer instead of setting it to nullptr)
+        // checks whether the String is empty
         explicit operator bool() const {
-            return buffer() != nullptr;
+            return length() != 0;
```

Comment:
?
I thought the idea was to check whether the string is a usable object. Original string (and anyone using ArduinoCore api stuff, which retains this behaviour) need this check since it *may* return null when calling c_str()
(...but, might as well have checked that directly...)

Reply:
We are currently incorrectly using it [by checking a String length](https://github.com/esp8266/Arduino/blob/1bb041bb6c488dd5e6462ab8b84a01fd2ebae168/libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.cpp#L324) which is the natural meaning of this operator no ?

Reply:
Sort of. Arduino api wrote it as a check for buffer() ptr being set, not length
Plus, operator[] includes both length and buffer() != null checks anyway

Reply:
I had missed the [official API updated version](https://github.com/arduino/ArduinoCore-avr/blob/bc792a9a716289249fa5f2f67be3ee8ba6507d6c/cores/arduino/WString.h#L138) (and https://www.artima.com/articles/the-safe-bool-idiom). 
This is far beyond this PR (and simplicity). I have reverted this change.

Reply:
Just to note, that was pre C++11. Article issue is resolved by `explicit operator bool() const`